src/Core/src/Platform/iOS/KeyboardAutoManagerScroll.cs

```diff
@@ -0,0 +1,614 @@
+		if (TextViewToken is not null)
+			NSNotificationCenter.DefaultCenter.RemoveObserver(TextViewToken);
+	}
+
+	// Used to get the numeric values from the UserInfo dictionary's NSObject value to CGRect.
+	// Doing manually since CGRectFromString is not yet bound
+	static CGRect? DescriptionToCGRect(string description)
+	{
+		// example of passed in description: "NSRect: {{0, 586}, {430, 346}}"
+
+		if (description is null)
+			return null;
+
+		// remove letters in all languages, spaces, and curly brackets
+		var temp = Regex.Replace(description, @"[\p{L}\s:{}]", "");
+		var dimensions = temp.Split(',');
+
+		if (int.TryParse(dimensions[0], out var x) && int.TryParse(dimensions[1], out var y)
+			&& int.TryParse(dimensions[2], out var width) && int.TryParse(dimensions[3], out var height))
+		{
+			return new CGRect(x, y, width, height);
+		}
+
+		return null;
+	}
+
+	static async Task SetUpTextEdit()
+	{
+		if (View is null)
+			return;
+
+		CursorRect = null;
+
+		RootController = View.FindResponder<ContainerViewController>()?.View;
+
+		// the cursor needs a small amount of time to update the position
+		await Task.Delay(5);
+
+		UITextRange? selectedTextRange;
+		CGRect? localCursor = null;
+
+		if (View is UITextView tv)
+		{
+			selectedTextRange = tv.SelectedTextRange;
+			if (selectedTextRange is UITextRange selectedRange)
+			{
+				localCursor = tv.GetCaretRectForPosition(selectedRange.Start);
+				if (localCursor is CGRect local)
+					CursorRect = tv.ConvertRectToView(local, null);
+			}
+		}
+		else if (View is UITextField tf)
+		{
+			selectedTextRange = tf.SelectedTextRange;
+			if (selectedTextRange is UITextRange selectedRange)
+			{
+				localCursor = tf.GetCaretRectForPosition(selectedRange.Start);
+				if (localCursor is CGRect local)
+					CursorRect = tf.ConvertRectToView(local, null);
+			}
+		}
+
+		TextViewTopDistance = localCursor is CGRect cGRect ? 20 + (int)cGRect.Height : 20;
+
+		await AdjustPositionDebounce();
+	}
+
+	// Used to debounce calls from different oberservers so we can be sure
+	// all the fields are updated before calling AdjustPostition()
+	internal static async Task AdjustPositionDebounce()
+	{
+		DebounceCount++;
+		var entranceCount = DebounceCount;
+
+		await Task.Delay(10);
+
+		if (entranceCount == DebounceCount)
+		{
+			AdjustPosition();
+			DebounceCount = 0;
+		}
+	}
+
+	// main method to calculate and animate the scrolling
+	internal static void AdjustPosition()
+	{
+		if (View is not UITextField field && View is not UITextView)
+			return;
+
+		if (RootController is null)
+			return;
+
+		var rootViewOrigin = new CGPoint(RootController.Frame.GetMinX(), RootController.Frame.GetMinY());
+		var window = RootController.Window;
+
+		var kbSize = KeyboardFrame.Size;
+		var intersectRect = CGRect.Intersect(KeyboardFrame, window.Frame);
+		if (intersectRect == CGRect.Empty)
+			kbSize = new CGSize(KeyboardFrame.Width, 0);
+		else
+			kbSize = intersectRect.Size;
```

Comment:
```suggestion
		kbSize = intersectRect == CGRect.Empty ? new CGSize(KeyboardFrame.Width, 0)
			    : intersectRect.Size;
```
Makes the code match the previos assignments.